Update ADL API versions and Account functionality

begoldsm opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 
### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).
### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/azure/azure-rest-api-specs/635)

<!-- Reviewable:end -->

begoldsm committed:
quick test removing transfer-encoding

begoldsm committed:
Rev all ADL swagger to GA API version.

Missed one spec.

Commit:
update ADLS swagger specs to include a "flush" api

Add list credential support back.

delete credential params not required

If the caller is the owner of the account they do not have to supply the
password parameter.

Commit:
Update firewall account properties.

Commit:
Updates for TrustedIdProvider

Address CR comments and change flush constant

properties are not read-only

they can be passed during create.

Commit:
Address lint issues to add descriptions.

begoldsm commented:
@amarzavery can you take a quick look and get this merged? Its needed for me to make code freeze for xplat cli.

Commit:
Updates based on CR feedback.

remove untrue consumes/produces. Start adding descriptions.

matt1883 committed:
Merge pull request #1 from begoldsm/master

Pulling latest

matt1883 committed:
Updated response descriptions

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+          "description": "the results of the list operation"
+        },
+        "count": {
+          "type": "integer",
```
is this valid? I have never seen any ARM RP sending more than value and nextLink in the output of pageable operations

begoldsm replied:
Agreed and removed.

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+        },
+        "maxDegreeOfParallelism": {
+          "type": "integer",
+          "format": "int32",
```
min/max ?
Is this a required property?

begoldsm replied:
Not required, I will add a min/max and default.

begoldsm replied:
fixed!

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+          "description": "the account regional location."
+        },
+        "name": {
+          "type": "string",
```
resdonly: true?

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+    },
+    "DataLakeAnalyticsAccount": {
+      "properties": {
+        "location": {
```
location should be required.

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+      },
+      "description": "Additional Azure Storage account parameters."
+    },
+    "DataLakeAnalyticsAccount": {
```
There should be a REsource object and DataLakeAnalyticsAccount should inherti (allof) from the Resource obhject. Take a look at the compute spec as an example.

begoldsm replied:
Is this critical to do now? I ask because based on compute I would need two objects for this, too:
Resource and CreateResource, where CreateResource has "location" as a required parameter.

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+        "name"
+      ],
+      "properties": {
+        "name": {
```
why should the name be required? It is provided in the path.

begoldsm replied:
Agreed and fixed!

amarzavery replied:
I still see name as a required property.

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DataLakeStoreAccountInfoProperties",
```
x-ms-client-flatten:true

begoldsm replied:
fixed!

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/StorageAccountProperties",
```
x-ms-client-flatten:true

begoldsm replied:
fixed!

amarzavery commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AddStorageAccountParameters"
```
PATCH operation cannot have required model properties. The model properties should not have a default value.

begoldsm replied:
I am confused here. In order to update the Storage account, you must pass in the storage account parameters, since we need the access key that you are using to update the storage account. Where is it a requirement that you can't have required parameters in the body in a patch?

amarzavery replied:
I am talking about the schema of AddStorageAccountParameters. This model has some required properties.

General PATCH semantics says that the body for PATCH should not have required parameters. Since PATCH is essentially  like a git diff. So only the things that one needs to change will be provided everything else remains constant. Over here having a required property in a model means you are forcing the customer to provide a value for a property even if he does not want to change that.

begoldsm replied:
I see, so I should create a new type of parameter called UpdateStorageAccountParameters that does not have any required properties. I can do that.

matt1883 committed:
Updated response descriptions

begoldsm committed:
Merge pull request #3 from matt1883/master

Updated response descriptions

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+          "Account"
+        ],
+        "operationId": "Account_DeleteStorageAccount",
+        "description": "Updates the specified Data Lake Analytics account to remove an Azure Storage account.",
```
This is actually correct. We are removing a storage account from an existing analytics account, so this is an update operation on the analytics account.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+        "tags": [
+          "Account"
+        ],
+        "operationId": "Account_GetDataLakeStoreAccount",
```
@amarzavery I will sync with you on this naming convention to understand better, because the name you are recommending above is more confusing. Today the objects are in the following namespace pattern:
Microsoft.Azure.Management.DataLake.Analytics.DataLake**AnalyticsAccountManagementClient**.Account

Your recommendation would result in something like:
Microsoft.Azure.Management.DataLake.Analytics.DataLakeAnalyticsAccountManagementClient.AnalyticsAccounts, which is redundant and doesn't actually differentiate anything, since you can't do anything with store accounts inside an analytics account client other than add/remove them from underneath the analytics account.

amarzavery replied:
I am talking wrt the script the customer writes.
var client = new DatalakeAnalyticsClient( ....);
client.AnalyticsAccounts.GetDataLakeStoreAccount();

begoldsm replied:
Makes sense. I have changed the operationId for these operations to be DataLakeStoreAccounts.Get();

begoldsm commented on arm-datalake-analytics/catalog/2016-11-01/swagger/catalog.json:
```diff
@@ -374,6 +370,90 @@
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
```
That is a good question, I will confirm what they are and add those in here.

begoldsm replied:
No max but I will set the min to 1, since anything less than that does not make sense. We will page for anything more than our max page count.

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+        "tags": [
+          "Job"
+        ],
+        "operationId": "Job_GetStatistics",
```
We were told to go back to singular at one point. Is the requirement going forward that it should be "Jobs", "Accounts", "Catalogs" and "Filesystems"?

amarzavery replied:
Ahh. I think only if it collides with a model name. Sorry I missed that condition. If there is a model named Job then please change the noun in the operationId to jobs.

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "format": "uuid",
```
It is better to be explicit here that it needs to be a GUID to ensure they don't pass in an invalid string. This ensures type validation during method call.

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+            "VertexResourceInUserFolder",
+            "JobManagerResourceInUserFolder",
+            "StatisticsResourceInUserFolder"
+
```
Done

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+          "format": "int64",
+          "description": "the timestamp (in ticks) for the yarn application executing the job. This value should not be set by the user and will be ignored if it is."
+        },
+        "compileMode": {
```
It is not required. If it is not specified the server will chose the best mode depending on the current server load and the job type. I will add this information.

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+      ],
+      "properties": {
+        "runtimeVersion": {
+          "type": "string",
```
Unfortunately not, it is a custom label string that is used for internal flighting, debugging and for private customer tests, the labels can grow forever, so maintaining this as an enum would become cost prohibitive.

amarzavery replied:
i see.

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+    },
+    "JobInformation": {
+      "properties": {
+        "jobId": {
```
I agree with your assessment. I will make this read-only to be explicit, since it is ignored if passed in the payload. This is a breaking change, though.

amarzavery replied:
not really. The service was royally ignoring it any ways. Now we are being explicit about it.

begoldsm replied:
Right, but anyone that set it in their code previously will get compilation errors, since it is no longer settable, whether the service was ignoring it or not. I agree that the number of people setting it should be small and we are also making a lot of other breaking changes :)

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+        },
+        "degreeOfParallelism": {
+          "type": "integer",
+          "format": "int32",
```
Yes I believe the default the server chooses is 1. Also I have added the minimum, the max can't be set here since it is based on a different account level setting, which is configurable by the account admin.

amarzavery replied:
If the default is 1 then please add `"default": 1`

begoldsm replied:
Done!

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+          "description": "the link (url) to the next page of results."
+        },
+        "count": {
+          "type": "integer",
```
We no longer return a count so I will remove this.

amarzavery replied:
please make sure to do so in all the models that are returned in a pageable result.

begoldsm replied:
Will do!

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeStore/accounts/{accountName}/firewallRules/{firewallRuleName}": {
```
I agree, this is a really good idea. I will make this change, but it won't make it for the current PowerShell build in november (which code freeze is today for).

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+        "tags": [
+          "Account"
+        ],
+        "operationId": "Account_DeleteFirewallRule",
```
The Noun should be in plural form only if there is a model that has the same name as the Noun in the operation id.

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+        }
+      }
+    },
+	"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeStore/accounts/{accountName}/trustedIdProviders/{trustedIdProviderName}": {
```
Why dont you use VS code with the linter plugin. makes life easy.
 https://github.com/Azure/openapi-lint-extension.

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+          },
+          "description": "The list of firewall rules associated with this Data Lake store account."
+        },
+		"trustedIdProviderState": {
```
indentation went awry

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+          "readOnly": true,
+          "description": "the link (url) to the next page of results."
+        },
+        "count": {
```
remove count

begoldsm commented on arm-datalake-analytics/job/2016-11-01/swagger/job.json:
```diff
@@ -0,0 +1,1070 @@
+            "modelAsString": false
+          }
+        },
+        "submitter": {
```
No it is read-only. I will add that.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The desired return format. Return the response in particular formatxii without access to request headers for standard content-type negotiation (e.g Orders?$format=json). Optional."
```
I have removed all of these, since they are not supported.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+  "definitions": {
+    "FirewallRuleProperties": {
+      "properties": {
+        "startIpAddress": {
```
Both are required, I have updated them to be reflected this way.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+          "description": "the namespace and type of the firewall Rule."
+        },
+        "id": {
+          "type": "string",
```
yup!

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+    "EncryptionIdentity": {
+      "properties": {
+        "type": {
+          "type": "string",
```
You don't have to pass in encryption information, but if you do certain things are required. I will add explicit requirements further down the property chain.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+      "description": "The encryption identity properties."
+    },
+    "EncryptionConfig": {
+      "properties": {
```
added in requirements for encryption if the object is created (the objects themselves are not required).

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+      "description": "Metadata information used by account encryption."
+    },
+    "DataLakeStoreAccountProperties": {
+      "properties": {
```
Nope everything is optional. I did add requirements in the nested properties, though, if they are passed.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+            "modelAsString": false
+          }
+        },
+		"firewallRules": {
```
I am going through visual studio code now to make sure indentation is good everywhere.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1311 @@
+      },
+      "description": "Data Lake Store error information"
+    },
+    "AzureAsyncOperationResult": {
```
No, I think it was a hold-over from hyak. It can be removed since there is a new way to handle LROs

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+        "tags": [
+          "Account"
+        ],
+        "operationId": "Account_GetStorageAccount",
```
I changed this the same way you recommended in store:
StorageAccounts_Get

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DataLakeAnalyticsAccount"
```
I can make it required: false. Nothing in a DataLakeAnalyticsAccount is required today, so this should be fine. If we need to be more explicit about required properties for PUT operations in the future, we can do so.

begoldsm replied:
fixed, I ended up completely changing the PATCH for analytics accounts to use a different object.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+          "description": "the account name of the Data Lake Store account."
+        },
+        "properties": {
+          "$ref": "#/definitions/DataLakeStoreAccountInfoProperties",
```
added everywhere except for JobProperties, which as we discussed can't use it since it has a discriminator and that would be strange.

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -131,15 +87,29 @@
+            "description": "Indicates the data being sent to the server is being streamed in chunks.",
+            "enum": [
+              "chunked"
+            ]
```
No because this is a constant, so having a single enum results in AutoRest turning the value into a constant string. Unless that changed and we need to use x-ms-enum now, but we have been just using "enum" for the past several releases for these things.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+      },
+      "description": "Generic resource error information."
+    },
+    "AzureAsyncOperationResult": {
```
No and it is removed :)

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -529,14 +527,11 @@
             "description": "The constant value for the operation.",
             "enum": [
-              "GETFILESTATUS"
+              "MSGETFILESTATUS"
```
No because they behave the same with some additional properties returned in MSGETFILESTATUS. The caller doesn't know the OP query param is different, they only know that they are getting more stuff back (nothing else is renamed).

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -868,14 +969,61 @@
+            "description": "The constant value for the operation.",
+            "enum": [
+              "REMOVEDEFAULTACL"
+            ]
```
same comment about use of single enum == constant in AutoRest. If this needs to change for some reason, please let me know and we can change them.

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -939,14 +1091,11 @@
             "description": "The constant value for the operation.",
             "enum": [
-              "GETACLSTATUS"
+              "MSGETACLSTATUS"
```
same comment as before.

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -1231,45 +1375,36 @@
           "readOnly": true,
           "format": "int64",
-          "description": "Gets the number of bytes used by the contet."
+          "description": "the number of bytes used by the contet."
```
Good find! Fixed.

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -1290,59 +1425,65 @@
+        },
+        "expirationTime": {
+          "type": "integer",
+          "readOnly": true,
```
It is only set using _SetExpiry which doesn't take this object in. I will make this a better description that indicates that if the value returned is 0, then there is no expiration.

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -1290,59 +1425,65 @@
-            "File",
-            "Directory"
+            "FILE",
+            "DIRECTORY"
```
Correct, this change was made a long time ago, though. I missed it during my copy paste (if you look in the old API version it is already in all caps, and that change was announced and versioned and everything).

begoldsm commented on arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json:
```diff
@@ -1419,30 +1521,128 @@
+          "$ref":  "#/definitions/AdlsRemoteException" 
+        } 
+      ],
+      "description": "A WebHDFS exception thrown indicating that one more arguments is incorrect. Thrown when a 400 error response code is returned (bad request)."
```
According to the WebHDFS spec "IllegalArgumentExceptions" are only thrown with a corresponding 400 error. If our service does not honor the webhdfs spec here, its a bug (same with the other comments about specific return codes).

Commit:
Changes to address PR comments for ADL.

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	arm-datalake-store/account/2016-11-01/swagger/account.json

Commit:
Minor fixes for PATCH

Commit:
A couple updates based on best practices

add "ByAccount" to sub lists. Updated documentation for CreateOrUpdate.

Commit:
More best practices for ADL swagger

remove 404 as a success case.
Add the ability to specify syncFlags for append calls.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1580 @@
+        },
+        "properties": {
+          "readOnly": true,
+          "$ref": "#/definitions/BlobContainerProperties",
```
fixed!

Commit:
fix some comments and fix storage patch.

this removes required parameters from patch for storage accounts.

Commit:
Added missing descriptions.

Commit:
Make naming consistent for top level updates.

Commit:
Missed reversion of pluralization.

Commit:
sub resource names cannot be read only

They are not sent with requests and if you include a sub resource with a
PUT on the top level resource it will fail since the name is empty.

johanste commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1597 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
This should either be a 204/No Body, or it SHOULD return the deleted/representation of resource (i.e. have a schema for the return value) for response code 200

[See: RFC2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.7/)

begoldsm commented:
@johanste @amarzavery if there are no other blocking issues, can this be merged? I am going to need a lot of time to make the necessary changes in both powershell and xplat_cli to snap to these changes since there are significant breaks, and the sooner I can get started the better. Thanks!

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+            "schema": {
+              "$ref": "#/definitions/FirewallRule"
+            },
+            "description": "Parameters supplied to create the create firewall rule."
```
Parameters supplied to create **or update** the ~~create~~ firewall rule

begoldsm replied:
fixed!

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+      },
+      "description": "Data Lake Store firewall rule properties information"
+    },
+    "FirewallRule": {
```
"required": ["properties"]

begoldsm replied:
good catch, fixed!

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+            "modelAsString": false
+          }
+        },
+        "keyVaultMetaInfo": {
```
wouldnt it be weird to not ask the customer to provide keyVaultMetaInfo ?

begoldsm replied:
No, because in the case where our service manages the encryption, they don't supply any key vault information because our service uses an internal keyvault for them.

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+          "type": "string",
+          "description": "the status of the Data Lake Store account after provisioning has completed.",
+          "enum": [
+            "active",
```
This is so weird. "encryptionState" and "provisioningState" has enum values in PascalCase whereas state has all the values in lowercase. I hope this is the actual wire representation. It would be nice if a consistent approach is taken w.r.t enum values for the enitre RP (Azure wide would be even better).

begoldsm replied:
This was a bug on my side. The enums over the wire are all PascalCase. I have fixed this in the code to be consistent (everything starts with a capital letter now).

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+          "description": "The current state of encryption provisioning for this Data Lake store account.",
+          "enum": [
+            "Creating",
+            "Succeeded"
```
Are you sure you guys never fail ?

begoldsm replied:
This is an implementation choice from our service. The "Creating" state is a the non-success state. So if encryption can't be enabled for some reason, it will stay in the "Creating" state. When calling "enableKeyVault" method, that is when you will see an error as to why the provisioning state can't move to succeeded.

amarzavery commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+          "description": "the gateway host."
+        },
+        "defaultGroup": {
+          "type": "string",
```
is there a default value for this? What value does the server choose if this is not provided by the customer?

begoldsm replied:
the default is nothing (empty string). There shouldn't be anything set here by default. This is for the customer to optionally decide if they want to have a default group added with permissions to their filesystem. Nothing should be added if they don't specify anything.

amarzavery commented on arm-datalake-store/filesystem/2015-10-01-preview/swagger/filesystem.json:
```diff
@@ -555,6 +555,72 @@
         }
       }
     },
+    "/webhdfs/v1/{flushFilePath}": {
```
We will have to revisit this again. This is an invalid swagger spec from swagger specification point of view. We may need to correct this by putting all the overloaded paths inside x-ms-paths.

begoldsm replied:
I agree. I am also removing this specific API since it isn't actually functional :).

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1597 @@
+          "200": {
+            "description": "Successfully retrieved the specified Azure Storage container information.",
+            "schema": {
+              "$ref": "#/definitions/BlobContainer"
```
@amarzavery this is the current name for the object in our service as well and we don't currently have plans to allow for retrieval of all your Azure storage objects through the data lake SDK (since that can be done through the storage SDK). This is mostly to allow a quick "at a glance" look at the members of an attached storage account. I think the risk of extension to include tables and queues is fairly low. 

That being said, what name would you recommend? Just Blob?

begoldsm replied:
Renamed BlobContainer -> StorageContainer.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1597 @@
+          "200": {
+            "description": "Successfully retrieved the list of Azure Storage containers under the specified Azure Data Lake Analytics account.",
+            "schema": {
+              "$ref": "#/definitions/ListBlobContainersResult"
```
Ah I see, I can just rename both of these to "StorageContainer" if that is best?

begoldsm replied:
Renamed BlobContainer -> StorageContainer

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1597 @@
+    "CreateStorageAccountInfo": {
+      "required": [
+        "properties",
+        "name"
```
No it doesn't when you are adding it during an account creation PUT. Then you MUST include the name of the storage account, otherwise ADLA account creation will fail.

begoldsm commented on arm-datalake-analytics/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1597 @@
+          "description": "the state of the Data Lake Analytics account.",
+          "enum": [
+            "active",
+            "suspended"
```
Good catch, it should be in PascalCase (first letter upper case). Fixing this across the board.

begoldsm commented on arm-datalake-store/account/2016-11-01/swagger/account.json:
```diff
@@ -0,0 +1,1321 @@
+          "FirewallRules"
+        ],
+        "operationId": "FirewallRules_CreateOrUpdate",
+        "description": "Creates or updates the specified firewall rule. In this case of update, the firewall rule with the specified name will be replaced with this new firewall rule.",
```
fixed for both.

Commit:
Addressed PR comments

Change Blob -> Storage
Fix casing for enums
Fix descriptions
Remove "flush" API
Add syncFlag to create

Commit:
Update the documentation

Flush no longer exists and is not actually required.

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR635)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/869

AutorestCI commented:
No modification for Ruby

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-node.svg?branch=RestAPI-PR635)](https://travis-ci.org/AutorestCI/azure-sdk-for-node) https://github.com/Azure/azure-sdk-for-node/pull/1970
